File: app/src/main/java/com/firstapp/robinpc/tongue_twisters_deluxe/new_app/utils/RetrieveLevelFigures.java
package com.firstapp.robinpc.tongue_twisters_deluxe.new_app.utils;

import android.content.Context;
import android.support.v4.content.ContextCompat;

import com.firstapp.robinpc.tongue_twisters_deluxe.R;
import com.firstapp.robinpc.tongue_twisters_deluxe.new_app.data.LevelFigures;

import java.util.ArrayList;
import java.util.List;

public class RetrieveLevelFigures {

    private Context context;

    private List<LevelFigures> levelFiguresList = new ArrayList<>();
    private String[] levelNumberHeaders, levelNames;
    private int[] levelImages;

    public RetrieveLevelFigures(Context context) {
        this.context = context;
        fetchLevelFigures();
        addLevelFiguresToList();
    }

    private void fetchLevelFigures() {
        fetchLevelImages();
        fetchLevelNumberHeaders();
        fetchLevelNames();
    }

    private void fetchLevelImages() {
        levelImages = new int[] {
                R.drawable.level_one,
                R.drawable.level_two,
                R.drawable.level_three,
                R.drawable.level_four,
                R.drawable.level_five,
                R.drawable.level_six,
                R.drawable.level_seven,
                R.drawable.level_eight,
                R.drawable.level_nine,
                R.drawable.level_ten
        };
    }

    private void fetchLevelNumberHeaders() {
        levelNumberHeaders = context.getResources().getStringArray(R.array.level_number_headers);
    }

    private void fetchLevelNames() {
        levelNames = context.getResources().getStringArray(R.array.level_names);
    }

    private void addLevelFiguresToList() {
        for(int i=0; i<levelImages.length; i++) {
            addLevelFigureToList(levelImages[i], levelNumberHeaders[i], levelNames[i]);
        }
    }

    public List<LevelFigures> getLevelFiguresList() {
        return levelFiguresList;
    }

    private void addLevelFigureToList(int levelDrawable, String levelNumberHeader, String levelName) {
        levelFiguresList.add(new LevelFigures(levelDrawable, levelNumberHeader, levelName));
    }
}

File: app/src/main/java/com/firstapp/robinpc/tongue_twisters_deluxe/utils/RetrieveLevelFigures.kt
package com.firstapp.robinpc.tongue_twisters_deluxe.utils

import android.content.Context

import com.firstapp.robinpc.tongue_twisters_deluxe.R
import com.firstapp.robinpc.tongue_twisters_deluxe.data.LevelFigures

import java.util.ArrayList

class RetrieveLevelFigures(private val context: Context) {

    private val levelFiguresList = ArrayList<LevelFigures>()
    private var levelNumberHeaders: Array<String>? = null
    private var levelNames: Array<String>? = null
    private var levelImages: IntArray? = null

    init {
        fetchLevelFigures()
        addLevelFiguresToList()
    }

    private fun fetchLevelFigures() {
        fetchLevelImages()
        fetchLevelNumberHeaders()
        fetchLevelNames()
    }

    private fun fetchLevelImages() {
        levelImages = intArrayOf(NULL_EQUIVALENT_DRAWABLE_ID, R.drawable.level_one, R.drawable.level_two, R.drawable.level_three, R.drawable.level_four, R.drawable.level_five, R.drawable.level_six, R.drawable.level_seven, R.drawable.level_eight, R.drawable.level_nine, R.drawable.level_ten)
    }

    private fun fetchLevelNumberHeaders() {
        levelNumberHeaders = context.resources.getStringArray(R.array.level_number_headers)
    }

    private fun fetchLevelNames() {
        levelNames = context.resources.getStringArray(R.array.level_names)
    }

    private fun addLevelFiguresToList() {
        for (i in levelImages!!.indices) {
            addLevelFigureToList(levelImages!![i], levelNumberHeaders!![i], levelNames!![i])
        }
    }

    fun getLevelFiguresList(): List<LevelFigures> {
        return levelFiguresList
    }

    private fun addLevelFigureToList(levelDrawable: Int, levelNumberHeader: String, levelName: String) {
        levelFiguresList.add(LevelFigures(levelDrawable, levelNumberHeader, levelName))
    }

    companion object {
        const val NULL_EQUIVALENT_DRAWABLE_ID = -1
    }
}
